swri_route_util/src/util.cpp

```diff
@@ -709,6 +709,11 @@ bool extractSubroute(
   end_index++;
   end_index = std::min(end_index, route.points.size());
 
+  if (end_index <= start_index)
+  {
+    return true;
+  }
+
```

Comment:
Did you ask me to review this just to see if I was paying attention?  

I'm not sure I follow all of the logic here 100%, but or this part, does this actually constitute a success?  Or did you mean to return false. If it's fine, should you have cleared sub_route.points `sub_route.points.clear()` before returning here or up at the top of this function?

Reply:
I think it should return true.  We asked for an empty subroute and we return an empty subtoute.  Success!

The use case I'm thinking of is that a user requests a subroute that is [route start, route_end - 50m).  If the route is less than 50m then this would be an empty route, which would not be a failure and would be the intuitive result.